Doc/library/itertools.rst

```diff
@@ -474,7 +474,7 @@ loops that truncate the stream.
    If *start* is zero or ``None``, iteration starts at zero.  Otherwise,
    elements from the iterable are skipped until *start* is reached.
 
-   If *stop* is ``None``, iteration continues until the iterator is
+   If *stop* is ``None``, iteration continues until the iterable is
```

Comment:
I think "iterator" is the more accurate term here. `islice` takes an iterable as an argument but creates an iterator from it. It is then that iterator that is being exhausted, not the iterable (which may be a non-exhaustable iterable such as a list).

Reply:
Technically this is true, but *islice* only really makes sense for inputs that are iterators. Also from the user's point of view the input *iterable* which is usually an iterator is what they control.

Reply:
I see. But as you say, `islice` only really makes sense for inputs that are iterators, so I still think the term iterator would click better for the user. I certainly see where you're coming from, reusing the name of the parameter though. So yeah I guess it's fine either way.

Reply:
Note that the parameter to this function is also called "iterable" (https://docs.python.org/3/library/itertools.html#itertools.islice). That is another argument to me for using "iterable" here.

Reply:
@JelleZijlstra Yes, as I also noted in my last comment regarding reusing the name of the parameter. But also note that the first sentence of the doc for `islice` reads "Make an iterator that returns selected elements from the iterable.", so I don't think there would be any confusion when later in the paragraph the term "the iterator" is mentioned.